dnf-behave-tests/createrepo_c/steps/lib/repodata.py

```diff
@@ -55,3 +55,32 @@ def verify_repomd_item_with_file(item, filepath, decompressed_filepath):
     if (a < b - 1 or a > b + 1):
         raise AssertionError("Timestamp of File: " + filepath + " doesn't match"
                              "timestamp in the repomd.xml:\n" + str(a) + " vs. " + str(b))
+
+def keys_do_not_differ(prim, flist, oth):
+    if set(prim.keys()) != set(flist.keys()):
+        raise AssertionError("Primary and Filelists have different package sets.")
```

Comment:
These assertions will be rather unhelpful when someone starts getting them. One way to do it is include the difference in the assertion text itself. For larger data that are better printed in multiline and with something to highlight the differences to the reader this gets a bit impractical.

How about using my great `print_lines_diff()`? :slightly_smiling_face:
https://github.com/rpm-software-management/ci-dnf-stack/blob/a7aa1a30866ae5ae6a19ec0f4cc1de0a3f2467ee/dnf-behave-tests/common/lib/diff.py#L15

It's a bit makeshift (and I don't know how the data you are comparing look like, so it may not display them well), but it does the job in coloring the diff so it's easier to see what's actually different. You can tweak it too if you want; just make sure you don't break the current usage - because it only gets called when a scenario fails, so you won't know it unless you test it explicitly.

Reply:
The original lines are basically completely different, so `print_lines_diff` would almost always display everything red. But I could print which packages are missing from which file, that should be easy and potentially helpful.

Reply:
Ok.. I have very little idea of how the data you're working with here actually look like, but you could just print a diff of the keys you're comparing for the assert (`set(prim.keys()) != set(flist.keys())`), as that is what really matters and so that's what the person running the tests should see.

You can see I've added support for diffing lists and tuples to the `print_lines_diff` function some time ago, I think it can compare sets just the same, I just didn't add it as I didn't have a use for it.